utils/utils_tx.py

```diff
@@ -541,11 +542,12 @@ def get_event_from_tx(event_id: str, tx_hash: str, proxy: ProxyNetworkProvider)
     return event
 
 
-def get_deployed_address_from_tx(tx_hash: str, proxy: ProxyNetworkProvider) -> str:
-    event = get_event_from_tx("SCDeploy", tx_hash, proxy)
-    if event is None:
-        return ""
-    return event.address.to_bech32()
+
+def get_deployed_address_given_deployer(deployer: Account) -> str:
```

Comment:
I don't particularly like relying on this precalculated address rather than fetching it from the transaction itself due to the fact that it doesn't provide any indication whether the deploy really passed. We had too many surprises relying on the precalculated method only to find out too late that the original deploy transaction failed for whatever completely unexpected reason.
In the entire flow of the environment, relying blindly on the address without checking the operation can result in very bad configs.
Checking now would need to separately fetch the final status of the transaction and that is either painfully unreliable or time consuming, so in this case, if I have to have this check anyway because I cannot rely on the address only, I might as well directly rely on the `SCDeploy` event of the transaction itself which is a solid indicator whether it worked or not. If it didn't, program should just crash 'cause there's no reason to continue without it.
In the economics of actual code, I know it's time and resource consuming to fetch the tx again from the network after deploying it, but I have to do it in any case due to the above reasons. 
The only difference is whether I directly do it at the `atomic` deploy level or should I do this check separately every time I deploy something. Since I don't have cases in which I don't care whether it worked or not, I would rather enforce it at this level and save me the headache of forgetting the check and actually failing.

Reply:
This is unfortunately a no go for me in this state.